src/testFixtures/java/uk/gov/di/ipv/cri/common/library/aws/SQSHelper.java

```diff
@@ -173,118 +126,176 @@ public void deleteMessages(String queueUrl, List<Message> messages) {
      */
-    public void deleteMatchingMessages(
-            String queueUrl, int count, Map<String, String> filters, boolean deleteNonMatching)
-            throws InterruptedException {
-        deleteMessages(queueUrl, count, Objects.requireNonNull(filters), deleteNonMatching);
+    public int tryDeleteMatchingMessages(String queueUrl, int count, Map<String, String> filters) {
+        return tryDeleteMessages(queueUrl, count, Objects.requireNonNull(filters));
     }
 
-    private List<Message> receiveMessages(
-            String queueUrl, int count, Map<String, String> filters, boolean deleteNonMatching)
+    private int deleteMessages(
+            String queueUrl, int count, Map<String, String> filters, boolean throwOnTimeout)
             throws InterruptedException {
-        final FilteredMessages messages = getMessages(queueUrl, count, filters);
+        final var messages = receiveMessages(queueUrl, count, filters, throwOnTimeout);
+        deleteMessagesFromList(queueUrl, messages);
+        return messages.size();
+    }
+
+    private int tryDeleteMessages(String queueUrl, int count, Map<String, String> filters) {
+        int deletedMessageCount;
 
-        if (deleteNonMatching) {
-            deleteMessagesFromList(queueUrl, messages.nonMatchingMessages);
-            resetMessageVisibility(queueUrl, messages.matchingMessages);
-        } else {
-            resetMessageVisibility(queueUrl, messages.allMessages());
+        try {
+            deletedMessageCount = deleteMessages(queueUrl, count, filters, false);
+        } catch (InterruptedException e) {
+            return 0;
         }
 
-        return messages.matchingMessages;
+        return deletedMessageCount;
     }
 
-    private void deleteMessages(
-            String queueUrl, int count, Map<String, String> filters, boolean deleteNonMatching)
+    private List<Message> receiveMessages(
+            String queueUrl, int count, Map<String, String> filters, boolean throwOnTimeout)
             throws InterruptedException {
-        final FilteredMessages messages = getMessages(queueUrl, count, filters);
+        final var messages = getMessages(queueUrl, count, filters);
 
-        if (deleteNonMatching) {
-            deleteMessagesFromList(queueUrl, messages.allMessages());
-            return;
+        if (messages.size() < count && throwOnTimeout) {
+            throw new InterruptedException(
+                    String.format("Received %d/%d messages", messages.size(), count));
         }
 
-        deleteMessagesFromList(queueUrl, messages.matchingMessages);
-
-        if (filters != null) {
-            resetMessageVisibility(queueUrl, messages.nonMatchingMessages);
-        }
+        return messages;
     }
 
-    private FilteredMessages getMessages(String queueUrl, int count, Map<String, String> filters)
-            throws InterruptedException {
+    private List<Message> getMessages(String queueUrl, int count, Map<String, String> filters) {
```

Comment:
Could you split this method up so it's more readable? As it stands, it's hard to understand the logic and is creating a bit of cognitive load. Thanks

Reply:
I was thinking the same - I should be able to make the inner loop a method

Reply:
Done ✅